pkg/network/node/node.go

```diff
@@ -311,38 +311,36 @@ func (node *OsdnNode) validateMTU() error {
 		return fmt.Errorf("unable to determine MTU while performing validation")
 	}
 
-	if int(node.networkInfo.MTU) > mtu+50 {
+	const MTUTaintKey string = "network.openshift.io/mtu-too-small"
+	n, err := node.kClient.CoreV1().Nodes().Get(node.hostName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not get Kubernetes Node object by hostname: %v", err)
+	}
+	mtuTooSmallTaint := &corev1.Taint{Key: MTUTaintKey, Value: "value", Effect: "NoSchedule"}
+	tainted := taints.TaintExists(n.Spec.Taints, mtuTooSmallTaint)
+	mtuWithOverhead := mtu + 50
+	if !tainted && int(node.networkInfo.MTU) > mtuWithOverhead {
```

Comment:
Sigh. This check was wrong to begin with. Ricky, you're excused because you're new to openshift-sdn, but @squeed and I both suck for not catching it in the original PR.

`node.networkInfo.MTU` is the VXLAN MTU. `mtu` is the actual MTU of the node's eth0. In order for the VXLAN to work, the node's MTU must be at least 50 bytes larger than the VXLAN MTU. So the correct check is:

    needsTaint := mtu < int(node.NetworkInfo.MTU) + 50

(As opposed to the current code which adds 50 to the wrong side, and mis-handles the `==` case.)

QE didn't catch this because their tests were focusing more on the 9001-vs-1500 case, and as it happens, 8951 is greater than either 1450 or 1550.

Reply:
sigh, gotcha...

Reply:
damn damn damn. I stared at that for a while and convinced myself it was right, too.